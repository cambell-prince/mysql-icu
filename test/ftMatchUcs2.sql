-- Test full text parsing
set names utf8;

use mysql_icu_test;

drop table if exists matchTest;
drop table if exists matchTest2;

create table matchTest  (
    id INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    phrase VARCHAR(128)
) collate ucs2_icu_my_MM_ci;
create table matchTest2  (
    id INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    phrase VARCHAR(128)
) collate ucs2_icu_custom_ci;

-- Fulltext indexes are not supported with UCS2, only boolean mode is supported

insert into matchTest2 (phrase) values 
('မြန်မာစကား'),
('စကာလုံးများ'),
('ဘယ်နှလုံးလဲ။'),
('ဘာလုပ်မလဲ။'),
('မြန်မာဘာသာစကား'),
('မြန်မာလူမျို'),
('မြန်မာနိုင်ငံ'),
('နိုင်ငံရေး'),
('လူများတယ်'),
('အင်္ဂလိန်ကမြန်မာပြည်ကိုလားတယ်။'),
('မြန်လားမာလား။');

insert into matchTest2 (phrase) values 
('This is English'),
('It is easier to use full text searches in English');

insert into matchTest select * from matchTest2;

repair table matchTest quick;
repair table matchTest2 quick;

-- Boolean mode can be used, but the parser can't be set!
select 'မြန်မာ*';
select * from matchTest where match (phrase) against ('မြန်မာ*' IN BOOLEAN MODE);
select 'လူ';
select * from matchTest where match (phrase) against ('လူ' IN BOOLEAN MODE);
select '+လူ';
select * from matchTest where match (phrase) against ('+လူ' IN BOOLEAN MODE);

select 'Custom rules';
select 'လုံး';
select * from matchTest2 where match (phrase) against ('လုံး' IN BOOLEAN MODE);
select '+လူ';
select * from matchTest2 where match (phrase) against ('+လူ' IN BOOLEAN MODE);

select 'မြန်မာစကား';
select * from matchTest2 where match (phrase) against ('မြန်မာစကား' IN BOOLEAN MODE);

select '+လူ -"မြန်မာ"';
select * from matchTest2 where match (phrase) against ('+လူ -မြန်' IN BOOLEAN MODE);

select '"မြန်မာ" boolean';
select * from matchTest2 where match (phrase) against ('"မြန်မာ"' IN BOOLEAN MODE);
select 'မြန်မာ*';
select * from matchTest2 where match (phrase) against ('မြန်မာ*' IN BOOLEAN MODE);

