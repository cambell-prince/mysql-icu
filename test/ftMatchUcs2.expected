Table	Op	Msg_type	Msg_text
mysql_icu_test.matchTest	repair	status	OK
Table	Op	Msg_type	Msg_text
mysql_icu_test.matchTest2	repair	status	OK
မြန်မာ*
မြန်မာ*
လူ
လူ
+လူ
+လူ
Custom rules
Custom rules
လုံး
လုံး
+လူ
+လူ
မြန်မာစကား
မြန်မာစကား
id	phrase
1	မြန်မာစကား
+လူ -"မြန်မာ"
+လူ -"မြန်မာ"
"မြန်မာ" boolean
"မြန်မာ" boolean
မြန်မာ*
မြန်မာ*
